src/visualization/types.jl

```diff
@@ -348,58 +348,77 @@ end
 
 # specializes the PlotData2D constructor to return an PlotData2DTriangulated for any type of mesh.
 function PlotData2DTriangulated(u, mesh, equations, dg::DGSEM, cache;
-                                solution_variables=nothing, nvisnodes=2*polydeg(dg))
+                                solution_variables=nothing, nvisnodes=nothing,
+                                elevations=nothing, slice=:yz, point=[0,0,0])
 
-  @assert ndims(mesh) == 2 "Input must be two-dimensional."
+  # convert variables based on solution_variables mapping
```

Comment:
Would you mind trying whether splitting the body of the function into two smaller parts for 2D/3D has any impact on initial latency, runtime performance, readability?